test/fw/ptl/utils/pbs_crayutils.py

```diff
@@ -141,3 +141,36 @@ def count_node_state(self, state='UP  B'):
             if stat['State'] == state:
                 count += 1
         return count
+
+    def get_numthreads(self, nid):
+        """
+        Returns the number of hyperthread for the given node
+        """
+        options = '-N %d -n -f "nid,c/cu"' % int(nid)
+        hostname = socket.gethostname()
+        platform = self.du.get_platform(hostname)
+        apstat_env = os.environ
+        apstat_cmd = "apstat"
+        if 'cray' not in platform:
+            return NULL
+        if 'craysim' in platform:
+            lib_path = '$LD_LIBRARY_PATH:/opt/alps/tester/usr/lib/'
+            apstat_env['LD_LIBRARY_PATH'] = lib_path
+            apstat_env['ALPS_CONFIG_FILE'] = '/opt/alps/tester/alps.conf'
+            apstat_env['apsched_sharedDir'] = '/opt/alps/tester/'
+            apstat_cmd = "/opt/alps/tester/usr/bin/apstat -d ."
+        cmd_run = self.du.run_cmd(hostname, [apstat_cmd, options],
+                                  as_script=True, wait_on_script=True,
+                                  env=apstat_env)
+        cmd_result = cmd_run['out']
```

Comment:
To get numthreads why not just get the first element of cmd_run['out'] and split it, for example:
    cmd_run['out'][1].split()[1]
but this can be made easier to read.

Reply:
That would be much less code.  But since I am not sure that the output of apstat is a stable interface, the existing code will handle future output changes a little better.

Reply:
This is fine.